src/file_pipe_log/log_file.rs

```diff
@@ -110,25 +183,44 @@ impl<F: FileSystem> LogFileWriter<F> {
     }
 }
 
+/// Build a file reader.
+///
+/// Attention please, the reader do not need a specified `[LogFileHeader]` from
+/// users.
+///
+/// * `[handle]`: standard handle of a log file.
+/// * `[expected_version]`: if `[None]`, reloads the log file header and parse
+/// the relevant `Version` before building the `reader`.
 pub(super) fn build_file_reader<F: FileSystem>(
     system: &F,
     handle: Arc<F::Handle>,
+    expected_version: Option<Version>,
 ) -> Result<LogFileReader<F>> {
     let reader = system.new_reader(handle.clone())?;
-    LogFileReader::open(handle, reader)
+    if let Some(version) = expected_version {
+        LogFileReader::open(handle, reader, version)
+    } else {
+        LogFileReader::open(
+            handle.clone(),
+            reader,
+            (build_file_header(system, handle)?).version(),
```

Comment:
`build_file_header` shouldn't recreate a fs reader. In most context the reader is ready. Here you should reuse the reader created by `system.new_reader(handle.clone())?;`.

Reply:
Done.